extensions/transferdetails/service.go

```diff
@@ -0,0 +1,284 @@
+package transferdetails
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/extensions"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	logging "github.com/ipfs/go-log"
+)
+
+// Service defines specific functions for extension funding
+type Service interface {
+	documents.Service
+
+	// DeriveFromUpdatePayload derives Funding from clientUpdatePayload
+	DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest) (documents.Model, error)
```

Comment:
The new api handler should be really simple, so we should move the existing logic from the handler. (calling derive and calling upate) to the service.


The service should only have public methods for the main interaction called by the new handler.

Like `Create` or `Update`. DeriveFromPayload methods should be private.

Reply:
following conversation with @mikiquantum , will leave as is.